docs/docsite/rst/guide_modulehelper.rst

```diff
@@ -0,0 +1,520 @@
+        )
+        use_old_vardict = False
+
+        def __run__(self):
+            self.vars.original_message = ''
+            self.vars.message = ''
+            if self.check_mode:
+                return
+            self.vars.original_message = self.vars.name
+            self.vars.message = 'goodbye'
+            self.changed = self.vars['new']
+            if self.vars.name == "fail me":
+                self.do_raise("You requested this to fail")
+
+
+    def main():
+        MyTest.execute()
+
+
+    if __name__ == '__main__':
+        main()
+
+
+Module Helper
+^^^^^^^^^^^^^
+
+Introduction
+""""""""""""
+
+``ModuleHelper`` is a wrapper around the standard ``AnsibleModule``, providing extra features and conveniences.
+The basic structure of a module using ``ModuleHelper`` is as shown in the `Quickstart`_ section above,
+but there are more elements that will take part in it.
+
+.. code-block:: python
+
+    from ansible_collections.community.general.plugins.module_utils.module_helper import ModuleHelper
+
+    class MyTest(ModuleHelper):
+        output_params = ()
+        change_params = ()
+        diff_params = ()
+        facts_name = None
+        facts_params = ()
+        use_old_vardict = True
+        mute_vardict_deprecation = False
+        module = dict(
+            argument_spec=dict(...),
+            # ...
+        )
+
+After importing the ``ModuleHelper`` class, you need to declare your own class extending it.
+
+.. seealso::
+
+    There is a variation called ``StateModuleHelper``, which builds on top
+    of the features provided by MH. See `StateModuleHelper`_ below for more details.
+
+The easiest way of specifying the module is to create the class variable ``module`` with a dictionary
+containing the exact arguments that would be passed as parameters to ``AnsibleModule``.
+If you prefer to create the ``AnsibleModule`` object yourself, just assign it to the ``module`` class variable.
+MH also accepts a parameter ``module`` in its constructor, if that parameter is used used,
+then it will override the class variable. The parameter can either be ``dict`` or ``AnsibleModule`` as well.
+
+Beyond the definition of the module, there are other variables that can be used to control aspects
+of MH's behavior. These variables should be set at the very beginning of the class, and their semantics are
+explained through this document.
+
+The main logic of MH happens in the ``ModuleHelper.run()`` method, which looks like:
+
+.. code-block:: python
+
+    @module_fails_on_exception
+    def run(self):
+        self.__init_module__()
+        self.__run__()
+        self.__quit_module__()
+        output = self.output
+        if 'failed' not in output:
+            output['failed'] = False
+        self.module.exit_json(changed=self.has_changed(), **output)
+
+The method ``ModuleHelper.__run__()`` must be implemented by the module and most modules will be able to perform their actions using only that method.
```

Comment:
I'd say 'many', not 'most'. As soon as a module has a certain complexity you don't want everything in one single method :)
```suggestion
The method ``ModuleHelper.__run__()`` must be implemented by the module and many modules will be able to perform their actions using only that method.
```

Reply:
I'll reword, because the idea was that _out of the override-able methods of MH_, most of the modules will only need to override that one. Now to the good practice of not cramming everything in one module, that's a whole other conversation.

Reply:
Better?

Reply:
Yes, thanks! :)